extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java

```diff
@@ -48,6 +49,9 @@ private void doHandle(RoutingContext ctx) {
         Context context = Vertx.currentContext();
         getHealth(reporter, ctx).emitOn(MutinyHelper.executor(context))
                 .subscribe().with(health -> {
+                    if (requestContext != null) {
```

Comment:
This should probably be in an `onTermination()`, or be duplicated in the failure callback

Reply:
Or, which is IMO the best solution, in an end handler added to the `RoutingContext`?

Reply:
I originally also used the end handler in my testing without issues.

Reply:
I think you can have only one endHandler, which means that if you have an interceptor adding one, you can have a problem. Note that this is only when you activate the scope in the code.

Reply:
Done.

Reply:
I think what @Ladicek did here make more sense as it makes it clear where the error handling is as opposed to having to look it up somewhere else

Reply:
If you guys think an end handler is a better solution, I'm totally fine with dropping this. Your call, I don't really know :-)